### PR TITLE
[admin] instrument check list

### PR DESCRIPTION
**Pull request by cmangeat**
base test
had to reorder models
remove back ref from restriction area to role
add a primary key on role_restrictionarea

need https://github.com/camptocamp/c2cgeoform/pull/118

### CODE REVIEW

**Comment by sbrunner** [comment]
Why all those changes in the model?

**Comment by cmangeat** [comment]
Changes have two purposes:

- reorder allow using table definitions  from another table definition, especially at colanderalchemy level, i.e. for widget association, it's for example easier to know Role when defining User -> such changes have no impact on data model (sql one).
- had to redefine "role_restrictionarea", adding one primary key instead of two in order to make checboxes widget work-> has an impact on data model, I plan to see whether it is possible to avoid it once test coverage will be better (one test more).

File: commons/c2cgeoportal_commons/models/main.py [hunk 1]
```diff
@@ -183,6 +183,72 @@ def __unicode__(self):
     schema=_schema
 )
 
+# association table role <> restriciton area
+class RAForRole(Base):
```
Comment (sbrunner) [inline commons/c2cgeoportal_commons/models/main.py]: Avoid using abbreviation =>
RestrictionArea_Role or RestrictionAreaForRole but less readable...

File: commons/c2cgeoportal_commons/models/main.py [hunk 2]
```diff
@@ -183,6 +183,72 @@ def __unicode__(self):
...
+    __tablename__ = 'role_restrictionarea'
+    __table_args__ = (
+        {"schema": _schema}
+    )
+
+    id = Column("id", Integer, primary_key=True)
+    role_id = Column("role_id", Integer, ForeignKey(_schema + ".role.id"))
+    restrictionarea_id = Column("restrictionarea_id", Integer, ForeignKey(_schema + ".restrictionarea.id"))
```
Comment (sbrunner) [inline commons/c2cgeoportal_commons/models/main.py]: Missing uncity chack on (role_id, restrictionarea_id)

File: commons/c2cgeoportal_commons/models/main.py [hunk 3]
```diff
@@ -183,6 +183,72 @@ def __unicode__(self):
...
+
+# association table layer <> restriciton area
+layer_ra = Table(
```
Comment (sbrunner) [inline commons/c2cgeoportal_commons/models/main.py]: Why moving this, is there any change?

File: commons/c2cgeoportal_commons/models/main.py [hunk 4]
```diff
@@ -183,6 +183,72 @@ def __unicode__(self):
...
+        ForeignKey(_schema + ".layer.id"), primary_key=True
+    ),
+    Column(
+        "restrictionarea_id", Integer,
+        ForeignKey(_schema + ".restrictionarea.id"), primary_key=True
+    ),
+    schema=_schema
+)
+
+class RestrictionArea(Base):
```
Comment (sbrunner) [inline commons/c2cgeoportal_commons/models/main.py]: Why moving this, is there any change?

**Comment by sbrunner** [comment]
Just for info it wasn't tow primary key but a primary key on two fields...

**Comment by sbrunner** [comment]
A migration script is required ...

File: commons/c2cgeoportal_commons/models/main.py (hunk 4 above)
Comment (cmangeat) [reply commons/c2cgeoportal_commons/models/main.py]: backref to role from restriction area was removed,
ref to RestricationArea added to Role:
https://github.com/camptocamp/c2cgeoportal/blob/a124ef3ff9752ab166f1bfc93313a15d7f519aa3/commons/c2cgeoportal_commons/models/main.py#L284

File: commons/c2cgeoportal_commons/models/main.py (hunk 3 above)
Comment (cmangeat) [reply commons/c2cgeoportal_commons/models/main.py]: RestrictionArea has to know layer_ra

https://github.com/camptocamp/c2cgeoportal/blob/a124ef3ff9752ab166f1bfc93313a15d7f519aa3/commons/c2cgeoportal_commons/models/main.py#L232

**Pull request closed without merging**